sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -508,24 +500,141 @@ export class EventHubSender extends LinkEntity {
     message: AmqpMessage | Buffer,
     options: SendOptions & EventHubProducerOptions = {}
   ): Promise<void> {
-
     const abortSignal: AbortSignalLike | undefined = options.abortSignal;
     const sendEventPromise = () =>
-      new Promise<void>((resolve, reject) => {
+      new Promise<void>(async (resolve, reject) => {
+        // Callbacks used for being registered with events on sender
```

Comment:
`rejectOnAbort `, `onAborted`, `removeListeners`  and `actionAfterTimeout` arent registered as listeners on the sender, but they do appear along the 4 that are registered right below this comment. So, this comment has the potential of being confusing.

Either group the actual listeners that are being registered together or remove this comment altogether.